Google-Summer-of-Code-2025.md

```diff
@@ -1221,52 +1259,211 @@ This project entails introducing a developer-only UI (as part of the developer o
+    - **Important**: The files above were produced using lint 4.2.2, but the latest (as of drafting these notes) is 8.2.0.
+      - You should aim to use the latest _compatible_ version of the lint tool (8.2.0 likely won't work due to requiring JDK 17, but you'll need to check this).
+      - Furthermore, different versions will have differing results. You should aim to find the _most_ number of issues possible (i.e. configure the tool and its arguments to catch as many problems as it's able to within the codebase).
+  - The general flow of the script will probably be to process the exemptions proto in order to generate an Android-lint compatible XML exemption file (to pass in), and then run the linter and interpret the results to determine a pass/fail for the overall process.
+    - The lint tool supports text-based output, XML, and SARIF. It's expected that the script will parse either XML or SARIF formats as they are strongly formatted whereas text may change in future versions of the tool.
+- Script changes:
+  - A new proto message needs to be added in script_exemptions.proto to properly represent exemption failures that are found by Android Lint.
+    - These exemptions should be designed in a way to ensure that they aren't fragile, i.e. prefer exemption specific categories of failures on a per-file basis rather than per-line. While not perfect, we ultimately aim to reach 0 exemptions in the long-term so per-file is a reasonable stop-gap (we use this approach in other places, though Android lint's exemptions are more elaborate since we need categorical exemptions).
+  - A new textproto under scripts/assets to contain the new exemptions (all existing failures should be exempted by default before fixes are checked in).
+  - A new script + tests for wrapping Android lint (no new utilities are expected to be needed at this stage, though it's possible we will need something for interacting with cmdline tools).
+    - The script will probably need to fail gracefully if it can't locate Android lint, e.g. because cmdline tools wasn't installed, with instructions on how to install it (e.g. using sdkmanager). This should only be necessary if it's not possible to bundle the tool with the script (such as by using the Maven dependency mentioned above).
+  - Changes to static_checks.sh and static_checks.yml to include the new lint script in each, respectively.
+- UI/domain changes:
+  - Unlikely that there are substantial changes here, but a lot of UI files will require changes as part of addressing the issues of the lint categories mentioned in the project description.
+
+</details>
+
+<details>
+<summary>Suggested PM demo points</summary>
+
+- Milestone 1: Demonstrate the script can be run locally to catch lint issues. Demonstrate that the script runs on develop, and is passing. Demonstrate that the script runs for pull requests and can catch a lint issue potentially being introduced by the pull request. Show the exemptions file being used.
```

Comment:
Why do you need the "potentially" in the second-last sentence? Not sure what that entails.

What does "Show the exemptions file being used" mean -- maybe say "Show that exemptions listed in the exemptions file are not reported."? (Or, should we still report them in the logs as FYIs, but make them non-blocking?)

Reply:
That sentence read weirdly--I revised it accordingly. I also adopted your suggestion for the exemptions bit as it more clearly explains the intention. As for reporting the exemptions, we should just ignore them as anything else is unnecessarily noisy (we'll know what the exemptions are from the textproto file in the same way as our other validation/enforcement scripts).